src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerProvider.cs

```diff
@@ -47,7 +47,7 @@ public ConsoleLoggerProvider(IOptionsMonitor<ConsoleLoggerOptions> options, IEnu
             _optionsReloadToken = _options.OnChange(ReloadLoggerOptions);
 
             _messageQueue = new ConsoleLoggerProcessor();
-            if (DoesConsoleSupportAnsi())
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows) || DoesConsoleSupportAnsi())
```

Comment:
We might as well remove the duplicate check in `DoesConsoleSupportAnsi()` below, since this the is the only caller of it. And then rename the method to indicate it is `Windows-only`.

```suggestion
            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows) || DoesWindowsConsoleSupportAnsi())
```

Reply:
Updated